osmaxx/excerptexport/models/export.py

```diff
@@ -80,11 +80,17 @@ def status_update_url(self):
 
     def set_and_handle_new_status(self, new_status, *, incoming_request):
         assert new_status in dict(self.STATUS_CHOICES)
+        if self.status == new_status and self.last_update_overdue():
+            new_status = self.FAILED
+
         if self.status != new_status:
             self.status = new_status
             self.save()
             self._handle_changed_status(incoming_request=incoming_request)
 
+    def last_update_overdue(self):
+        return (self.updated_at + EXTRACTION_PROCESSING_TIMEOUT_TIMEDELTA) < timezone.now()
+
```

Comment:
Currently, this we only call this method on exports in non-final statuses. But as the name of the method doesn't reflect that it's only applicable to those, the method should return `False` for exports in a final status, no matter how long ago their last update happened.